rules/scripts/sival_signature_test.bash

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+set -o pipefail
+
+FILES_TO_CHECK=(@FILES@)
+
+IS_NEGATIVE_TEST=@IS_NEGATIVE_TEST@
+OPENTITANTOOL=@OPENTITANTOOL@
+VERIFY_ARGS=(@VERIFY_ARGS@)
+ECDSA_KEY=@ECDSA_KEY@
+SPX_KEY=@SPX_KEY@
+KEY_ARGS=()
+RESULT_STATUS=0
+
+echo "___Starting signature verification & manifest key verification___"
+
+for file in "${FILES_TO_CHECK[@]}"
+do
+
+  echo "Checking signature of ${file}..."
+
+  # TEST 1: ECDSA key and/or SPX key integrity and signature verification
+
+  # If ECDSA key is provided, include it in the manifest update
+  if [[ ! -z "$ECDSA_KEY" ]]; then
+    KEY_ARGS+=("--ecdsa-key=${ECDSA_KEY}")
+  fi
+  
+  # If SPX key is provided, include it in the manifest update
+  if [[ ! -z "$SPX_KEY" ]]; then
+    KEY_ARGS+=("--spx-key=${SPX_KEY}")
+  fi
+
+  # If there are keys to verify, perform manifest update
+  if [[ ${#KEY_ARGS[@]} -ne 0 ]]; then
+    echo -n "Checking key material in ${file}..."
+        ${OPENTITANTOOL}  --rcfile= image manifest update \
+          "${KEY_ARGS[@]}" \
+          --output=/tmp/manifest_test_key_verify.$$ \
+          ${file}
+
+    # Perform manifest comparison
+    echo "Comparing dummy manifest with original file..."
+
+    if [[ "$IS_NEGATIVE_TEST" == "True" ]]; then
```

Comment:
Are we going to move the negative test checks to the very end?

I think the individual checks should always perform as they normally would:  ie:

```
Checking key material in foo.signed.bin: ok
Checking signature in foo.signed.bin: failed
```

And then at the end:
```
Negative test: inverting result
PASS
```